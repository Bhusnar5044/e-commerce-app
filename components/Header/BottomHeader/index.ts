export { BottomHeader as default } from './BottomHeader';
