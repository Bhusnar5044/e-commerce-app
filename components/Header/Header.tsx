import Link from 'next/link';
import NavbarActions from '../NavbarActions';
import { Container } from '../ui/Container/Container';

export const Header = () => {
    return (
        <div className="bg-blue-200 w-full h-18 bg-amazon_blue text-lightText fixed top-0 z-50">
            <Container>
                <div className="relative px-4 sm:px-6 lg:px-8 flex h-16 items-center">
                    {/* logo */}
                    <Link
                        href={'/'}
                        className="px-2 border border-transparent hover:border-white cursor-pointer duration-300 flex items-center justify-center h-[70%]"
                    >
                        <p className="font-bold text-xl">E-COMMERCE-APP</p>
                    </Link>

                    {/* cart */}
                    <NavbarActions />
                </div>
            </Container>
        </div>
    );
};

export default Header;

Header.displayName = 'Header';
