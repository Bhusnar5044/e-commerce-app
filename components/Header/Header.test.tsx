import { renderWithStore, screen } from '@test-utils';
import { Header } from './Header';

jest.mock('next/navigation', () => ({
    ...require('next-router-mock'),
}));
describe('Header', () => {
    it('should render properly', () => {
        const { container } = renderWithStore(<Header />);
        expect(container).toMatchSnapshot();
        expect(screen.getByText(/E-COMMERCE-APP/)).toBeInTheDocument();
    });
});
