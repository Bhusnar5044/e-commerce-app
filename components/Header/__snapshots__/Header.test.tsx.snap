// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render properly 1`] = `
<div>
  <div
    class="bg-blue-200 w-full h-18 bg-amazon_blue text-lightText fixed top-0 z-50"
  >
    <div
      class="mx-auto max-w-7xl"
    >
      <div
        class="relative px-4 sm:px-6 lg:px-8 flex h-16 items-center"
      >
        <a
          class="px-2 border border-transparent hover:border-white cursor-pointer duration-300 flex items-center justify-center h-[70%]"
          href="/"
        >
          <p
            class="font-bold text-xl"
          >
            E-COMMERCE-APP
          </p>
        </a>
        <div
          class="ml-auto flex items-center gap-x-4"
        >
          <button
            class="w-auto border border-transparent disabled:cursor-not-allowed disabled:opacity-50 text-white font-semibold hover:opacity-75 transition flex items-center rounded-full bg-black px-4 py-2"
            type="button"
          >
            <svg
              class="lucide lucide-shopping-bag"
              fill="none"
              height="20"
              stroke="white"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M6 2 3 6v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2V6l-3-4Z"
              />
              <path
                d="M3 6h18"
              />
              <path
                d="M16 10a4 4 0 0 1-8 0"
              />
            </svg>
            <span
              class="ml-2 text-sm font-medium text-white"
            >
              0
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
